source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -110,6 +113,40 @@ namespace aspect
       return phi_theta_d;
     }
 
+    template <int dim>
+    Point<3>
+    EllipsoidalChunk<dim>::EllipsoidalChunkGeometry::push_forward_topography(const Point<3> &phi_theta_d_hat) const
+    {
+      const double d_hat = phi_theta_d_hat[2]; // long, lat, depth
+      Point<dim-1> phi_theta;
+      const double rad_to_degree = 180/numbers::PI;
+      if (dim == 3)
+        phi_theta = Point<dim-1>(phi_theta_d_hat[0] * rad_to_degree,phi_theta_d_hat[1] * rad_to_degree);
+      const double h     = topography->value(phi_theta);
+      const double d = d_hat + (d_hat + bottom_depth)/bottom_depth*h;
```

Comment:
Please add a comment to this line explaining what happens here. It looks like h is a relative value that is scaled by something (d_hat+bottom)/bottom, why is that the case?